### PR TITLE
Clean up extension and add Services.yaml

### PR DESCRIPTION
**Pull request by jschlier**

### CODE REVIEW

File: Classes/Event/CreatingPaginationEvent.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace B13\Newspage\Event;
+
+use Psr\EventDispatcher\StoppableEventInterface;
+use TYPO3\CMS\Core\Pagination\PaginationInterface;
+use TYPO3\CMS\Core\Pagination\PaginatorInterface;
+
+final class CreatingPaginationEvent implements StoppableEventInterface
+{
+    private PaginatorInterface $paginator;
+    private PaginationInterface $pagination;
```
Comment (ervaude) [inline Classes/Event/CreatingPaginationEvent.php]: Please set initial value to `null` as this is not initialized in the constructor.

```
private ?PaginationInterface $pagination = null;
```

File: Classes/Event/CreatingPaginationEvent.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->paginator = $paginator;
+    }
+
+    public function getPaginator(): PaginatorInterface
+    {
+        return $this->paginator;
+    }
+
+    public function getPagination(): PaginationInterface
```
Comment (ervaude) [inline Classes/Event/CreatingPaginationEvent.php]: Could also return `null` if Pagination was not set yet.

File: Classes/Event/CreatingPaginationEvent.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function setPagination(PaginationInterface $pagination): void
+    {
+        $this->pagination = $pagination;
+    }
+
+    public function isPropagationStopped(): bool
+    {
+        return isset($this->pagination);
```
Comment (ervaude) [inline Classes/Event/CreatingPaginationEvent.php]: Check for $this->pagination instanceof PaginationInterface

File: ext_localconf.php [hunk 4]
```diff
@@ -49,28 +49,30 @@
...
-    ];
-    $iconRegistry = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Imaging\IconRegistry::class);
-    foreach ($icons as $identifier => $path) {
-        if (!$iconRegistry->isRegistered($identifier)) {
-            $iconRegistry->registerIcon(
-                $identifier,
-                \TYPO3\CMS\Core\Imaging\IconProvider\SvgIconProvider::class,
-                ['source' => 'EXT:newspage/Resources/Public/Icons/' . $path . '.svg']
-            );
+    if (version_compare(TYPO3_version, '11.0.0', '<')) {
```
Comment (ervaude) [inline ext_localconf.php]: Please use `\TYPO3\CMS\Core\Information\Typo3Version`for this

**Review by ervaude — CHANGES_REQUESTED** [review]

**Comment by jschlier** [comment]
I included your suggestions @ervaude and also caught the pagination being null in the controller. Shouldn't happen, but the event listener _could_ be unset by configuration ?

Much better. Thanks!

**Pull request merged**